### PR TITLE
Display custom headers in compose

### PR DESCRIPTION
**Pull request by matthewhughes934**
* **What does this PR do?**

Allows displaying customer headers under a "Headers: " prompt in the compose screen

* **Screenshots (if relevant)**

Appearance with no custom headers (unchanged):

```
        From: 
          To: 
          Cc:
         Bcc:
     Subject: 
    Reply-To:
         Fcc: 
```

Appearance with a single custom header:

```
        From:
          To:
          Cc:
         Bcc:
     Subject:
    Reply-To:
         Fcc:
     Headers: X-First-Header: 1

```

Appearance with five custom headers:

```
        From:
          To:
          Cc:
         Bcc:
     Subject:
    Reply-To:
         Fcc:
     Headers: X-First-Header: 1
              X-Second-Header: 2
              X-Third-Header: 3
              X-Fouth-Header: 4
              X-Fifth-Header: 5
```

Appearance with more than 5 custom headers:

```
        From:
          To:
          Cc:
         Bcc:
     Subject:
    Reply-To:
         Fcc:
     Headers: X-First-Header: 1
              X-Second-Header: 2
              X-Third-Header: 3
              X-Fouth-Header: 4
              ...
```

* **Does this PR meet the acceptance criteria?**

   - Documentation created/updated (you have to edit
     [doc/manual.xml.head](https://www.github.com/neomutt/neomutt/blob/master/doc/manual.xml.head)
     for that) - not sure if these docs need updating

* **What are the relevant issue numbers?**

#2582

### CODE REVIEW

File: compose.c [hunk 1]
```diff
@@ -862,6 +902,42 @@ static int draw_envelope_addr(int field, struct AddressList *al,
...
+{
+  const char *overflow_text = "...";
+  int rows_used = 0;
+
+  struct ListNode *first = STAILQ_FIRST(&rd->email->env->userhdrs);
+  if (first)
+  {
+    /* Draw first entry on same line as prompt */
+    draw_header(rd->win_envelope, row, HDR_CUSTOM_HEADERS);
+    mutt_paddstr(rd->win_envelope->state.cols -
```
Comment (matthewhughes934) [inline compose.c:915]: I feel like there's definitely a simpler solution here. With the line above printing:
```
        Headers: 
```
I'm just wanting to add `first->data` directly after this, but truncated if wider than the window width

**Comment by matthewhughes934** [comment]
I'm not too familiar with `ncurses` or most of what mutt's libs have to offer, so please don't hesitate to point out where things can be done more cleanly :)

File: compose.c (hunk 1 above)
Comment (flatcap) [reply compose.c:915]: That looks right.
`draw_header()` manages the colour and alignment, so that needs to be separate.

**Comment by flatcap** [comment]
> I'm not too familiar with `ncurses` or most of what mutt's libs have to offer

We need a Window API -- a neat set of drawing primitives -- but I've been putting it off.
Ideas welcome.

File: compose.c [hunk 2]
```diff
@@ -862,6 +902,42 @@ static int draw_envelope_addr(int field, struct AddressList *al,
   return lines_used;
 }
 
+static int draw_envelope_user_hdrs(const struct ComposeRedrawData *rd, int row)
```
Comment (flatcap) [inline compose.c:905]: Please can you add a Doxygen comment block for this function.

File: compose.c [hunk 3]
```diff
@@ -862,6 +902,42 @@ static int draw_envelope_addr(int field, struct AddressList *al,
...
+{
+  const char *overflow_text = "...";
+  int rows_used = 0;
+
+  struct ListNode *first = STAILQ_FIRST(&rd->email->env->userhdrs);
+  if (first)
```
Comment (flatcap) [inline compose.c:911]: Rather than a function-spanning `if`, return early here.

File: compose.c [hunk 4]
```diff
@@ -862,6 +902,42 @@ static int draw_envelope_addr(int field, struct AddressList *al,
...
+    /* Draw first entry on same line as prompt */
+    draw_header(rd->win_envelope, row, HDR_CUSTOM_HEADERS);
+    mutt_paddstr(rd->win_envelope->state.cols -
+                     (HeaderPadding[HDR_CUSTOM_HEADERS] +
+                      mutt_strwidth(_(Prompts[HDR_CUSTOM_HEADERS]))),
+                 first->data);
+    rows_used++;
+
+    struct ListNode *np = STAILQ_NEXT(first, entries);
+    if (np)
```
Comment (flatcap) [inline compose.c:922]: We can return early here, too.

**Review by flatcap — CHANGES_REQUESTED** [review]
Looks good.  Just a few minor suggestions.

File: compose.c (hunk 3 above)
Comment (matthewhughes934) [reply compose.c]: Thanks, done with 89659fe

File: compose.c (hunk 2 above)
Comment (matthewhughes934) [reply compose.c:910]: Not sure how I missed that one, added with 4736e68

**Comment by matthewhughes934** [comment]
Let me know when you've had a chance to look over the latest changes, and if they look good I'll rebase.

**Review by flatcap — APPROVED** [review]
LGTM!

**Comment by matthewhughes934** [comment]
> LGTM!

Cheers, rebased and pushed :+1:

**Pull request merged**